PyRoute/TradeCalculation.py

```diff
@@ -691,6 +696,21 @@ def route_distance(route):
             start = end
         return distance
 
+    def route_cost(self, route):
+        """
+        Given a route, return its total cost via _compensated_ summation
```

Comment:
Using Kahan compensated summation here is an attempt to mitigate floating-point error arising from adding tiny values to huge ones (such as an accumulating route total).